implementations/synthetic_datagen/evolve_instruct/evolve_instruct_generator.py

```diff
@@ -0,0 +1,269 @@
+
+#Given Prompt#:
+<PROMPT>
+"""
+        )
+
+        self.prompt_templates[Mutation.ADD_CONSTRAINTS] = (
+            self.prompt_templates['base']
+            + f"""Add a few more constraints or requirements \
+                to #Given Prompt#, and create #New Prompt#. \
+                {write_in_korean}
+
+#Given Prompt#:
+<PROMPT>
+"""
+        )
+
+        self.prompt_templates[Mutation.DEEPEN] = (
+            self.prompt_templates['base']
+            + f"""Slightly increase the depth and breadth \
+                of #Given Prompt#, and create #New Prompt#. \
+                {write_in_korean}
+
+#Given Prompt#:
+<PROMPT>
+"""
+        )
+
+        self.prompt_templates[Mutation.CONCRETIZE] = (
+            self.prompt_templates['base']
+            + f"""Make #Given Prompt# slightly more concrete, \
+            and create #New Prompt#. {write_in_korean}
+
+#Given Prompt#:
+<PROMPT>
+"""
+        )
+
+        self.prompt_templates[Mutation.INCREASE_REASONING] = (
+            self.prompt_templates['base']
+            + f"""If #Given Prompt# can be solved with just a \
+            few simple thinking processes, rewrite it to explicitly \
+                request multi-step reasoning, and create #New Prompt#. \
+                {write_in_korean}
+
+#Given Prompt#:
+<PROMPT>
+"""
+        )
+
+        self.prompt_templates[Mutation.SWITCH_TOPIC] = (
+            self.prompt_templates['base']
+            + f"""Rewrite #Given Prompt# by switching the topic, \
+            keeping the domain and difficulty level similar, \
+                and create #New Prompt#. {write_in_korean}
+
+#Given Prompt#:
+<PROMPT>
+"""
+        )
+
+    def generate(self):
+        """
+        Generate synthetic instructions and instances.
+
+        This method triggers the generation of synthetic instructions
+        followed by
+        the generation of synthetic instances based on those instructions.
+        """
+        logging.info("Generating synthetic instructions...")
+        self.instruction_generator.generate()
+        logging.info("Generating synthetic instances...")
+        self.instance_generator.generate()
+
+    def evaluate(self):
+        raise RuntimeError(
+            "Evaluation not implemented for EvolveInstructGenerator yet "
+            " - use an LLM to evaluate the quality of the generations."
+        )
+
+    def run(self):
+        self.create_seed_prompts()
+        self.create_prompts()
+        self.create_answers()
+
+        list_qa = []
+        for i in range(len(self.final_prompts)):
+            if len(self.final_answers[i]) > 10:
+                list_qa.append(
+                    {
+                        'input': self.final_prompts[i],
+                        'output': self.final_answers[i],
+                    }
+                )
+        import json
+        import uuid
+
+        with open(
+            f"{
+                self.seed_data.replace('.jsonl', '').replace('json', '')
```

Comment:
The definition of self.seed_data in the constructor of the class is missing, this can be confusing